[7/N] Remove fork tests for RPC.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63454
* __->__ #63443

After https://github.com/pytorch/pytorch/pull/63442, all distributed
tests can run with opt-asan. As a result, we can now remove all of our fork
based tests.

This is the first PR in a stack, which first removes fork based tests from RPC.

Differential Revision: [D30384905](https://our.internmc.facebook.com/intern/diff/D30384905/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30384905/)!

pritamdamania committed:
[7/N] Remove fork tests for RPC.

After https://github.com/pytorch/pytorch/pull/63442, all distributed
tests can run with opt-asan. As a result, we can now remove all of our fork
based tests.

This is the first PR in a stack, which first removes fork based tests from RPC.

Differential Revision: [D30384905](https://our.internmc.facebook.com/intern/diff/D30384905/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30384905/)!

[ghstack-poisoned]

lw commented on torch/testing/_internal/distributed/rpc_utils.py:
```diff
@@ -75,26 +74,13 @@ def _check_and_unset_tcp_init():
-# - and subprocesses launched with either fork or spawn.
 # To avoid a combinatorial explosion in code size, and to prevent forgetting to
 # add a combination, these are generated automatically by the code in this file.
 # Here, we collect all the test suites that we need to cover and the two multi-
 # processing methods. We then have one separate file for each agent, from which
```
Remove "and the two multiprocessing methods".

lw commented on torch/testing/_internal/distributed/rpc_utils.py:
```diff
@@ -75,26 +74,13 @@ def _check_and_unset_tcp_init():
 # Here, we collect all the test suites that we need to cover and the two multi-
 # processing methods. We then have one separate file for each agent, from which
 # we call the generate_tests function of this file, passing to it a fixture for
 # the agent, which then gets mixed-in with each test suite and each mp method.
```
Remove "and each mp method".

lw commented on torch/testing/_internal/distributed/rpc_utils.py:
```diff
@@ -175,44 +150,32 @@ def generate_tests(
     module. The name of each test will be a concatenation of the `prefix` arg,
-    the original name of the test suite, and a suffix of either `WithFork` or
-    `WithSpawn`. The `module_name` should be the name of the calling module so
+    the original name of the test suite.
```
Add "and" before "the original name".

lw reviewed (approved):
There's some lint issues, and some other mentions of the MP helpers to remove. Other than that LGTM!

pritamdamania committed:
Update on "[7/N] Remove fork tests for RPC."


After https://github.com/pytorch/pytorch/pull/63442, all distributed
tests can run with opt-asan. As a result, we can now remove all of our fork
based tests.

This is the first PR in a stack, which first removes fork based tests from RPC.

Differential Revision: [D30384905](https://our.internmc.facebook.com/intern/diff/D30384905/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30384905/)!

[ghstack-poisoned]
